code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+		COMSIG_LIVING_STATUS_SLEEP, // Slime was asleep.
+		COMSIG_LIVING_STATUS_STAGGER), // Slime was staggered.
+		PROC_REF(call_end_ability))
+	RegisterSignal(attached_atom, COMSIG_PARENT_QDELETING, PROC_REF(call_end_ability)) // Attached atom was qdeleted.
+	RegisterSignal(attached_atom, COMSIG_ATOM_EX_ACT, PROC_REF(call_attached_atom_ex_act)) // Attached atom was hit by an explosion.
+	RegisterSignal(attached_atom, COMSIG_ATOM_BULLET_ACT, PROC_REF(call_attached_atom_projectile_hit)) // Attached atom was hit by a projectile.
+	if(isliving(attached_atom))
+		var/mob/living/attached_living = attached_atom
+		attached_living.buckle_mob(owner, TRUE, TRUE, 90)
+		attached_living.add_movespeed_modifier(MOVESPEED_ID_SLIME_ATTACHED, TRUE, 0, NONE, TRUE, SLIME_POUNCE_ATTACHED_SLOWDOWN) // An attached Slime will weigh down its victim.
+		RegisterSignal(attached_living, list(
+			COMSIG_MOVABLE_UNBUCKLE, // Slime was unbuckled.
+			COMSIG_MOB_DEATH, // Attached victim died.
+			COMSIG_LIVING_STATUS_UNCONSCIOUS), // Attached victim was unconscious.
+			PROC_REF(call_end_ability))
+		RegisterSignal(attached_living, COMSIG_LIVING_DO_RESIST, PROC_REF(call_living_resist)) // Attached victim used the Resist action.
+		if(passive_pounce)
+			return
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
+		if(xeno_owner.issamexenohive(attached_living) || HAS_TRAIT(attached_living, TRAIT_INTOXICATION_IMMUNE))
+			return
+		effect_over_time(SLIME_POUNCE_INTOXICATION_STACKS + xeno_owner.xeno_caste.additional_stacks) // Effects applied every few seconds to a Slime's victim.
+		return
+	previous_turf = get_turf(owner)
+	ADD_TRAIT(owner, TRAIT_IMMOBILE, TRAIT_GENERIC)
+	owner.forceMove(hit_atom)
+	owner.anchored = TRUE
+
+/// Called whenever the Pounce throw_at is completed, regardless of resolution.
+/datum/action/xeno_action/activable/slime_pounce/proc/throw_complete()
+	SIGNAL_HANDLER
+	UnregisterSignal(owner, list(COMSIG_MOVABLE_IMPACT, COMSIG_MOVABLE_POST_THROW))
+
+/// Applies effects over a set amount of time.
+/// Will call itself again based on SLIME_POUNCE_INFLICTION_RATE unless there is no longer an attached atom.
+/datum/action/xeno_action/activable/slime_pounce/proc/effect_over_time(stacks)
+	if(!attached_atom || !owner.buckled)
+		return
+	addtimer(CALLBACK(src, PROC_REF(effect_over_time), stacks), SLIME_POUNCE_INFLICTION_RATE)
+	var/mob/living/attached_living = attached_atom
+	if(attached_living.has_status_effect(STATUS_EFFECT_INTOXICATED))
+		var/datum/status_effect/stacking/intoxicated/debuff = attached_living.has_status_effect(STATUS_EFFECT_INTOXICATED)
+		debuff.add_stacks(stacks)
+		return
+	attached_living.apply_status_effect(STATUS_EFFECT_INTOXICATED, stacks)
+
+/// Calls attached_atom_ex_act via INVOKE_ASYNC.
+/datum/action/xeno_action/activable/slime_pounce/proc/call_attached_atom_ex_act(severity, epicenter_dist, impact_range)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(attached_atom_ex_act), severity, epicenter_dist, impact_range)
```

Comment:
Doesnt this make them get hit by the same explosion twice?
I get it if just calls endability but the proc it calls deals damage too

Reply:
Good catch, actually. I meant for that to counteract getting what is essentially god mode for being inside a wall, and is not intented to be applied in the case of living mobs and objects.
I've made it so that it will only apply to walls/turfs.